src/cgame/cg_main.cpp

```diff
@@ -777,74 +740,93 @@ enum {
 	LOAD_UPGRADES,
 	LOAD_CLASSES,
 	LOAD_BUILDINGS,
+	LOAD_VOICES,
+	LOAD_CLIENTS,
 	LOAD_REMAINING,
 	LOAD_DONE
 } typedef cgLoadingStep_t;
 
-static void CG_UpdateLoadingStep( cgLoadingStep_t step )
+/*
+======================
+CG_UpdateLoadingProgress
+
+======================
+*/
+
+static void CG_UpdateLoadingProgress( int step, const char *label )
 {
-#if 0
-	static int startTime = 0;
-	static int lastStepTime = 0;
-	const int thisStepTime = trap_Milliseconds();
+	cg.loadingFraction = ( 1.0f * step ) / LOAD_DONE;
 
-	switch (step) {
-		case LOAD_START:
-			startTime = thisStepTime;
-			Log::Notice("^4%%^5 Start loading.");
-			break;
-		case LOAD_DONE:
-			Log::Notice("^4%%^5 Done loading everything after %is (%ims).",
-					(thisStepTime - startTime)/1000, (thisStepTime - startTime));
-			break;
-		default:
-			Log::Notice("^4%%^5 Done with Step %i after %is (%ims)… Starting Step %i",
-					step - 1, (thisStepTime - lastStepTime)/1000, (thisStepTime - lastStepTime), step );
-			break;
+	Q_strncpyz( cg.loadingText, label, sizeof( cg.loadingText ) );
+
+	Log::Notice( "Loading: %d%% %s", (int) ( 100 * cg.loadingFraction), cg.loadingText );
```

Comment:
This seems too noisy for a Notice. Also `static_cast<int>`

Reply:
> This seems too noisy for a Notice.

Hmm, the purpose of funny messages is that they are expected to be read…

Reply:
I did the `static_cast<int>` thing. For the Notice, if I set a Debug nothing is printed while I have a lot of loglevel set to debug (including `logs.logLevel.default`)…

Reply:
To be read on the loading screen, not in the log file!

The level here is `logs.logLevel.cgame.default`

Reply:
how about using `G_LogPrintf` to have it in the console only?

Reply:
Console logs are for debugging. Funny messages don't help with that. The `Log::Notice` here should be removed